Bug #495: Activation Range

blunted2night opened the pull request:
This a reworked patch for fixing issue 495
https://bugs.openmw.org/issues/495

blunted2night committed:
split MWWord::World::update into multiple functions

blunted2night committed:
updated MWWorld::PhysicsSystem::getFacedXXX functions

 * changed the names and return values to be consistent
 * made the distance to search a parameter
 * change the distance returned to world units instead of percentage of query distance

blunted2night committed:
update MWWord::World to track distances while performing occlusion queries.

This allows checking the activation distance against what is stored in GameSettings. Experimentation suggests that a second distance is required for NPCs.

blunted2night committed:
Factored faced object lookups into MWRender::World

 * Renamed MWWorld::World::getFacedHandle to getFacedObject.
 * Changed it to return an object pointer
 * Updated clients to use return object directly.

blunted2night committed:
added a separate activation distance for NPCs

zinnschlag commented:
Can we have the additional command line option as discussed on the issue tracker, please? scrawl is right in that the correct activation range would be a major annoyance during testing.

blunted2night committed:
added option to override the activation distance

The command line option '--activate-dist <int>' can be used to override
the in game activation distance.
